### PR TITLE
[WOR-1693] Add swat test to verify Sam resource type definitions

### PR DESCRIPTION
**Pull request by marctalbott**
Ticket: [WOR-1693](https://broadworkbench.atlassian.net/browse/WOR-1693)
* These aren't all the actions that are included in each role. Instead, I selected the list based on what the current automated tests are checking as well as a few other actions that Rawls checks. I largely omitted actions that Leo uses as those are best tested by Leo.

---

**PR checklist**

- [ ] Include the JIRA issue number in the PR description and title
- [ ] Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] If you changed anything in `model/`, then you should [publish a new official `rawls-model`](https://github.com/broadinstitute/rawls/blob/develop/README.md#publish-rawls-model) and update `rawls-model` in [Orchestration's dependencies](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/project/Dependencies.scala).
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green, including CI tests
- [ ] **Squash commits and merge** to develop (branches are automatically deleted after merging)
- [ ] Inform other teams of any substantial changes via Slack and/or email

[WOR-1693]: https://broadworkbench.atlassian.net/browse/WOR-1693?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by marctalbott** [comment]
Tests will fail until https://github.com/broadinstitute/workbench-libs/pull/1695 is merged and the new version is used here

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala [hunk 1]
```diff
@@ -149,6 +149,51 @@ class RawlsApiSpec
...
+
+      resourceTypes.collect {
+        case ResourceType(typeName, roles, _, _, _, _) if typeName.equals("workspace") =>
+          roles.collect {
+            case ResourceRole(roleName, actions, descendantRoles, _) if roleName.equals("owner") =>
+              actions should contain allElementsOf(List("share_policy::reader", "share_policy::share-reader", "share_policy::writer", "share_policy::share-writer", "share_policy::can-compute", "share_policy::owner", "write", "own", "compute", "delete", "read_auth_domain", "update_auth_domain"))
+              descendantRoles.get("google-project") shouldBe Option(Set("owner"))
+            case ResourceRole(roleName, actions, descendantRoles, _) if roleName.equals("reader") =>
+              actions should contain allElementsOf(List("read", "read_policy::owner", "read_auth_domain"))
+              actions should not contain allElementsOf(List("write", "compute"))
```
Comment (cahrens) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala]: anyElementsOf? That is, neither "write" nor "compute" should be in the list of actions.

Comment (marctalbott) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala]: Yeah exactly

Comment (cahrens) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala]: But is there a checker for that? I would expect that "should not contain allElementsOf" tests that BOTH "write" and "compute" aren't in the list… not that neither of them is.

Comment (marctalbott) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala]: Ohh I understand, your expectation was correct. Fixed it by switching to `noElementsOf`

**Review by aherbst-broad — APPROVED** [review]

**Review by cahrens — APPROVED** [review]

**Pull request merged**